runtime/tr.source/trj9/codegen/J9TreeEvaluator.cpp

```diff
@@ -218,6 +218,11 @@ uint32_t J9::TreeEvaluator::calculateInstanceOfOrCheckCastSequences(TR::Node *in
    bool isInstanceOf = instanceOfOrCheckCastNode->getOpCodeValue() == TR::instanceof;
    bool mayBeNull = !instanceOfOrCheckCastNode->isReferenceNonNull() && !objectNode->isNonNull();
 
+   // By default maxOnsiteCacheSlotForInstanceOf is set to 0 which means cache is disable.
+   // To enable test pass JIT option maxOnsiteCacheSlotForInstanceOf=<number_of_slots>
+   bool createDynamicCacheTests = cg->comp()->getOptions()->getMaxOnsiteCacheSlotForInstanceOf() != 0;
```

Comment:
This comment may become stale very quickly if the value changes in the future. I don't really see the value of it.

Reply:
Well to avoid using another option to guard this changes, I added this check here. If the value is non-zero  `(should be greater than zero - I should change condition)` either set by default or passed as JIT option, we will generate tests accordingly.
Do you have other suggestions in mind? We do need number of onsite cache option to vary number of cache slots

Reply:
I think you may have misunderstood. IMO the comment provides no value and I think we should remove it. My justification for this is that if someone modifies the default value of `_maxOnsiteCacheSlotForInstanceOf` in the future they will also have the _know_ to update this comment which seems unlikely to happen; thus the comment becomes stale and incorrect and a future reader of the code may make incorrect assumptions.

Infact `getMaxOnsiteCacheSlotForInstanceOf` should be properly documented to inform the consumer of the API what the semantics of the return value is; i.e. a return value of zero means this feature should be disabled. My point is that this documentation and comment should be coupled with the `getMaxOnsiteCacheSlotForInstanceOf` function, not consumers of the function as found here.